### PR TITLE
Crl number 20 octets max

### PR DESCRIPTION
**Pull request by lolepezy**
Allow only CRL number that fit into 20 octets.

### CODE REVIEW

File: src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java [hunk 1]
```diff
@@ -96,6 +96,11 @@ public DateTime getNextUpdateTime() {
     }
 
     public X509CrlBuilder withNumber(BigInteger number) {
+        // CRL number must be representable in 20 octets
+        // https://tools.ietf.org/html/rfc5280#section-5.2.3
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java:100]: Maybe add this as javadoc on the method itself.

File: src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java [hunk 2]
```diff
@@ -96,6 +96,11 @@ public DateTime getNextUpdateTime() {
...
+        if (number.abs().bitLength() > 20 * 8) {
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java:101]: The actual range for CRLNumber is `0..MAX` where `MAX` is 20 octets. Probably should check the lower bound too.

**Review by bjpbakker — APPROVED** [review]

File: src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java (hunk 2 above)
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java]: Indeed!

File: src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java (hunk 1 above)
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlBuilder.java]: Yeap

**Pull request merged**